wordpress-mobile/WordPress-Android
# Issue/2789 add locale to wpcom api calls

Fixes #2789 

Outstanding points:
+ moved getRestLocaleParams into RestClientUtils where it belongs, plus takes locale from current configuration rather than asking Locale.getDefault which is only updated upon app launch

+ placed the call everywhere where needed so errors can come localised from the API

+ added URL sanitizer so we can add query params in the querystring and we make sure in the end the proper URL is composed within the request to the server

This last one might be controversial, as really there are so many parts in the code where URLs are formed by just using string concatenation 😱 , that it would be really hard to change all that and test everything, so I added a convenience method that will take (almost) whatever you throw at it and make a proper URL out of it. The thing we _should_ be doing is actually having one way of writing URLs throughout all of our codebase, but OTOH this allows us to use different techniques for treating URLs and it actually seems to work great FWIW 😄 .


To test:
Just use the app, specially go to Stats and other sections where the WP.com REST api is used, and you should check that the `&locale=` value currently configured is sent



Needs review: @maxme
